src/main/kotlin/Actions/CreateToDoAction.kt

```diff
@@ -1,23 +1,44 @@
 package Actions
 
-import Todo.ToDo
+import Errors.BadRequestError
+import Errors.UnsupportedMediaError
+import Todo.ToDoList
 import contenttype.HttpContentTypes
 import httpstatus.HttpStatus
+import request.Request
 import response.ResponseBuilder
 
-class CreateToDoAction(private val toDo: ToDo): Action {
+class CreateToDoAction(private val toDoList: ToDoList): Action {
     private val headers = mapOf("Content-Type" to (HttpContentTypes.JSON.type + HttpContentTypes.JSON.parameter))
     private lateinit var requestBody: String
 
-    override fun act(responseBuilder: ResponseBuilder): String {
-        val responseBody = toDo.createToDo(requestBody)
-        responseBuilder.setBody(responseBody)
-        responseBuilder.setHeaders(headers)
-        responseBuilder.setStatusCode(HttpStatus.Created)
-        return responseBuilder.build()
+    override fun act(responseBuilder: ResponseBuilder, request: Request): String {
+        return if (validContentType(request) && validBody(request) && toDoList.addTask(requestBody)) {
+            responseBuilder.setBody(requestBody)
+            responseBuilder.setHeaders(headers)
+            responseBuilder.setStatusCode(HttpStatus.Created)
+            responseBuilder.build()
+        } else if (validContentType(request) && (!validBody(request) || !toDoList.addTask(requestBody))) {
+            BadRequestError().handleError(responseBuilder)
+        } else {
+            UnsupportedMediaError().handleError(responseBuilder)
+        }
```

Comment:
Nitpick: I find it a bit hard to understand all the conditions lead to this `else` block. My understanding is that if the content type is invalid, you want to return the `UnsupportedMediaError`. I personally think it's easier to understand if you put that at the very top by itself (between line 15 and 16).

```
if (!validContentType(request)) {
  return UnsupportedMediaError().handleError(responseBuilder)
}
```

By having one condition it's really easy to reason it out.

Reply:
I agree! By doing so, it also reduces the number of times I'm calling `toDoList.addTask(requestBody)`